pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -273,6 +288,13 @@ func RunStartBuild(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra
 	return exitErr
 }
 
+// isBuildLogTimeout returns true if the error indicates timeout getting build
+// logs (default is 10 seconds).
+// FIXME: This is ugly and should be replaced by checking the error type.
+func isBuildLogTimeout(err error) bool {
+	return strings.Contains(err.Error(), "timed out")
```

Comment:
I still hate this. Can't you somehow check for the [code](https://github.com/kubernetes/kubernetes/blob/55f5e48047a1c3e38b917703153e5bf4e2dcb2e6/pkg/api/errors/errors.go#L261) that is added in the timeout error? Maybe try to cast to StatusError?